src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/HMPPSAuthService.kt

```diff
@@ -2,36 +2,62 @@ package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
 
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Service
 import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.util.UriComponentsBuilder
+import reactor.core.publisher.Mono
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthGroupID
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
 
-val AuthServiceProviderGroupPrefix = "INT_SP_"
+class UnverifiedEmailException : RuntimeException()
 
-data class AuthGroup(
+data class UserDetail(
+  val firstName: String,
+  val email: String,
+)
+
+private const val AuthServiceProviderGroupPrefix = "INT_SP_"
+
+private data class AuthGroupResponse(
   val groupCode: String,
   val groupName: String,
 )
 
+private data class AuthUserDetailResponse(
+  val firstName: String,
+  val email: String,
+  val verified: Boolean,
+)
+
+private data class UserEmailResponse(
+  val email: String,
+)
+
+private data class UserDetailResponse(
+  val name: String,
+)
+
 @Service
 class HMPPSAuthService(
-  @Value("\${hmppsauth.api.locations.user-groups}") private val userGroupsLocation: String,
+  @Value("\${hmppsauth.api.locations.auth-user-groups}") private val authUserGroupsLocation: String,
+  @Value("\${hmppsauth.api.locations.auth-user-detail}") private val authUserDetailLocation: String,
+  @Value("\${hmppsauth.api.locations.user-email}") private val userEmailLocation: String,
+  @Value("\${hmppsauth.api.locations.user-detail}") private val userDetailLocation: String,
   private val hmppsAuthApiWebClient: WebClient,
 ) {
   fun getServiceProviderOrganizationForUser(user: AuthUser): AuthGroupID? {
     if (user.authSource != "auth") {
       return null
     }
 
-    val url = UriComponentsBuilder.fromPath(userGroupsLocation)
+    val url = UriComponentsBuilder.fromPath(authUserGroupsLocation)
       .buildAndExpand(user.userName)
       .toString()
 
     val groups = hmppsAuthApiWebClient.get().uri(url)
       .retrieve()
-      .bodyToFlux(AuthGroup::class.java)
+      .bodyToFlux(AuthGroupResponse::class.java)
       .collectList().block()
 
     val serviceProviderOrgs = groups
```

Comment:
not something you have changed, but unless we are wanting to explicitly state that a service provider organisation is a group (which we may?), can this not just be assigned to the above request?